### PR TITLE
DOC: Improve links and fix issues on resample documentation page

### LINKED ISSUE #26275: Links on resample documentation page are broken
I meant to open an issue about inconsistent behaviour of resampling for sub- and super-daily intervals. When I was almost done, I discovered it's already been reported/discussed #9586

Figuring all of this out myself wasted a couple of hours, that could have been saved by a clearer documentation on the `resample` page: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.resample.html

Some things to change here:
* The links under *Notes* are "broken" in a sense that the headers on the page they link to seem to be non-existing anymore. I guess the link labeled *this link* should go here: http://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#dateoffset-objects  I'm unsure to which header *user guide* should link to.
* The docstring for the `rule` parameter is confusing. It's type annotation says it's supposed to be a `str` but the description is *The offset string or object representing target conversion.* What kind of object can it be if not a string? And shouldn't this be in the type annotation then? Is it possible to link to some more information on this directly from the description of the parameter so it becomes immediately clear what the options are?

### PR DESCRIPTION
**Pull request by bharatr21**
- [x] closes #26275 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry (Not required IMO)

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -7924,7 +7924,7 @@ def resample(self, rule, how=None, axis=0, fill_method=None, closed=None,
 
         Parameters
         ----------
-        rule : str
+        rule : DateOffset object or str
```
Comment (jreback) [inline pandas/core/generic.py]: make this: DateOffset, Timedelta, or str

**Comment by jreback** [comment]
small comment, otherwise lgtm. ping on green.

**Comment by Bharat123rox** [comment]
The error is `AttributeError: type object 'DataFrame' has no attribute 'to_panel'` which seems to be unrelated to the current PR. @jreback Any suggestions?

**Comment by WillAyd** [comment]
@Bharat123rox I think #26308 will resolve that so can just merge master here after that goes in

**Comment by Bharat123rox** [comment]
Ping @jreback to review

**Comment by jreback** [comment]
thanks @Bharat123rox

**Pull request merged**